plugins/modules/syslog_server.py

```diff
@@ -0,0 +1,320 @@
+    if (
+        module.params["port"] != DEFAULT_PORT
+        and module.params["port"] != syslog_server.port
+    ):
+        payload["port"] = module.params["port"]
+    if module.params["protocol"]:
+        protocol = get_protocol(module.params["protocol"])
+        if protocol != UDP and protocol != syslog_server.protocol:
+            payload["protocol"] = protocol
+    return payload
+
+
+def update_syslog_server(module: AnsibleModule, rest_client: RestClient):
+    # Get record of old syslogServer by host
+    old_syserver = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+
+    before = old_syserver.to_ansible()
+    payload = build_update_payload(module, rest_client, old_syserver)
+
+    module.warn("payload: " + str(payload))
+
+    # Return if there are no changes
+    if payload == {
+        "host": before.get("host"),
+        "port": before.get("port"),
+        "protocol": before.get("protocol"),
+    }:
+        return False, before, dict(before=before, after=before)
+
+    # Otherwise, update with new email address
+    old_syserver.update(
+        rest_client=rest_client,
+        payload=payload,
+        check_mode=module.check_mode,
+    )
+
+    new_syserver = SyslogServer.get_by_host(
+        host=payload["host"], rest_client=rest_client
+    )
+    after = new_syserver.to_ansible()
+
+    return (
+        after != before,
+        after,
+        dict(before=before, after=after),
+    )  # changed, records, diff
+
+
+def delete_syslog_server(module: AnsibleModule, rest_client: RestClient):
+    delete_syserver = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+
+    if not delete_syserver:
+        return False, {}, dict(before={}, after={})
+
+    before = delete_syserver.to_ansible()
+    delete_syserver.delete(rest_client, module.check_mode)
+
+    return (
+        True,
+        {},
+        dict(before=before, after={}),
+    )  # changed, records, diff
+
+
+def run(module: AnsibleModule, rest_client: RestClient):
+    syslog_server = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+    state = module.params["state"]
+    if state == "present":
+        if syslog_server:
+            return update_syslog_server(module, rest_client)
+        return create_syslog_server(module, rest_client)
+
+    # Else if state == "absent":
+    return delete_syslog_server(module, rest_client)
+
+
+def validate_params(module):
+    params = []
+    if module.params["state"] != "present":
+        if module.params["host_new"] is not None:
+            params += "host_new"
+        if module.params["port"] is not None:
+            params += "port"
+        if module.params["protocol"] is not None:
+            params += "protocol"
+
+    if params:
+        msg = ", ".join(params) + " can be used only if state==present"
+        module.fail_json(msg=msg)
```

Comment:
I think the second example in `EXAMPLES` does not comply with this.

We need to decide if `port` and `protocol` can be changed at same time as `name`.
I do not see why that could not be possible, or why it would not be logical.
But I want this to be consistent with other modules - them `vm` module in particular.

What `vm` does - looking at integration test - ups, we do not test `vm_name_new` at all. :/.
Relevant code is https://github.com/ScaleComputing/HyperCoreAnsibleCollection/blob/main/plugins/modules/vm.py#L378-L394. A guess - we have bug in vm module, try to rename old_vm twice, on first call it gets renamed, but on second call we do not find renamed VM, and instead create a new VM with vm_name_new, so we end with 2 VMs with name==vm_name_new. But an integration test is needed to confirm this.

I cannot tell for sure if you should support changing name at same time as `port`, `protocol`. I think we will want to allow this. Please wait with this change until it is clear. Or maybe you have a different opinion?

Reply:
My opinion would be to allow changing ``port`` and ``protocol`` at the same time because, why not? Let's say a user has created a syslog server with a certain port and protocol, but then realized that the port and protocol he'd created were wrong and would like to change both of them. Would it not then be better for the user to be able to change both ``port`` and ``protocol`` in the same task (at the same time) than in two separate tasks? Just my view on this...

Also, I've used the ``user`` module to help with the implementation of this module.

Reply:
I agree with that - "to allow changing port and protocol at the same time".

Also VM module will be updated to allow changing name in same way as any other parameter we can change on VM. So we are consistent.